src/index.ts

```diff
@@ -14,4 +13,10 @@ export {
   responseInterface,
   CacheInterface
 } from './types'
-export default useSWR
+
+// A work around for the `__esModule` mark as microbundle
+// doesn't support mixed exports currently.
+// https://github.com/developit/microbundle/issues/306#issuecomment-459103476
+import * as exports from './use-swr'
+Object.defineProperty(exports, '__esModule', { value: true })
```

Comment:
looks like the variable `exports` is been resolved as another variable instead of cjs `exports` runtime variable by ts compiler.

// cjs bundle
```js
var E = s.Provider,
  x = { // the `exports` variable after compressed
    __proto__: null,
    trigger: R,
    mutate: j,
    SWRConfig: E,
    default: C
  };

...

Object.defineProperty(x, "__esModule", {
  value: !0
}), exports.SWRConfig = E, exports.cache ... // global exports is still there
```

we might need to find another workaround for it 😢

Reply:
WTH... can we bypass TS here?

Reply:
manually adding the esModule mark for cjs build in postbuild script seems to work https://github.com/vercel/swr/pull/919

Reply:
It feels too dirty, I wonder if we want to do that...

Reply:
removing main field and cjs bundle also works (for next.js case), since its webpack config on server mode can also include module field in mainFields.

but I don't know if there's other case that cjs bundle will be consumed only or it cannot be not widely adopted.

Reply:
few more updates to make this work:

* only keep `"module"` filed and add `type: "module"` can indicate the package as esm module.  reference: https://nodejs.org/api/packages.html#packages_type
* and rename `jest.config.js` to `jest.config.cjs` to still make jest run under commonjs
* also need to update changes from master (jest/ts-jest to 25.x, and few fixes for test) : )

I tried them in #919, tested with node v10 locally. it passed CI check with node v14. you can also rebase master then apply changes here, and close that one.

Reply:
I personally feel that CJS is still a must for supporting v0.x (related: #163). But we can probably land this in v1.0... 💭